src/masternodes/loan.h

```diff
@@ -260,17 +300,62 @@ inline CInterestRateV2 ConvertInterestRateToV2(const CInterestRate& rate1)
     return rate2;
 }
 
+inline CInterestRateV2 ConvertInterestRateToV2(const CInterestRateV3& rate3)
+{
+    CInterestRateV2 rate2{};
+    rate2.height = rate3.height;
+    rate2.interestPerBlock = rate3.interestPerBlock.amount;
+    rate2.interestToHeight = rate3.interestToHeight.amount;
+
+    return rate2;
+}
+
+inline CInterestRateV3 ConvertInterestRateToV3(const CInterestRate& rate1)
+{
+    CInterestRateV3 rate3{};
+    rate3.height = rate1.height;
+    rate3.interestPerBlock.amount = rate1.interestPerBlock;
+    rate3.interestToHeight.amount = rate1.interestToHeight;
+
+    return rate3;
+}
+
+inline CInterestRateV3 ConvertInterestRateToV3(const CInterestRateV2& rate2)
+{
+    CInterestRateV3 rate3{};
+    rate3.height = rate2.height;
+    rate3.interestPerBlock = {false, rate2.interestPerBlock};
+    rate3.interestToHeight = {false, rate2.interestToHeight};
+
+    return rate3;
+}
+
+inline const auto InterestCalculation = [](CNegativeInterest &a, const CNegativeInterest &b) {
+    if ((!a.negative && !b.negative) ||
+        (a.negative && b.negative)) {
+        a.amount += b.amount;
+    } else {
+        if (a.amount > b.amount) {
+            a.amount -= b.amount;
+        } else {
+            a.amount = b.amount - a.amount;
+            a.negative = !a.negative && b.negative;
```

Comment:
effectively `a.negative = b.negative`  would make it also easier to read

Reply:
Agreed.